src/cephadm/cephadmlib/systemd_unit.py

```diff
@@ -2,22 +2,138 @@
+    ) -> 'SystemdDropIn':
+        dropin = SystemdDropIn()
+        dropin.contents = data
+        return dropin
+
+    def add_section(self, section: str) -> None:
+        if section not in self.contents.keys():
+            self.contents[section] = {}
+
+    def add_setting(self, section: str, setting: str, value: str) -> None:
+        if section not in self.contents.keys():
+            raise SystemdDropInException(
+                f'section {section} not present in drop-in'
+            )
+
+        if setting not in self.contents[section]:
+            self.contents[section][setting] = []
+        self.contents[section][setting].append(value)
+
+    def rm_section(self, section: str) -> None:
+        if section in self.contents.keys():
+            del self.contents[section]
+
+    def rm_setting(
+        self, section: str, setting: str, value: Optional[str] = None
+    ) -> None:
+        if section not in self.contents.keys():
+            raise SystemdDropInException(
+                f'section {section} not present in drop-in'
+            )
+        # our drop-ins could have multiple values for the same setting in the same
+        # section e.g.
+        #
+        # [Unit]
+        # Wants = unit1
+        # Wants = unit2
+        #
+        # so we need to be able to either remove a specific value if it was
+        # passed or remove all values if nothing is specified
+        if setting in self.contents[section] and not value:
+            del self.contents[section][setting]
+        elif setting in self.contents[section] and value:
+            try:
+                self.contents[section][setting].remove(value)
+            except ValueError:
+                raise SystemdDropInException(
+                    f'Section {section} setting {setting} does not have value {value}'
+                )
+            if len(self.contents[section][setting]) == 0:
+                del self.contents[section][setting]
+
+    def to_str(self) -> str:
+        drop_in_text = '# generated by cephadm\n'
+        for section, settings in self.contents.items():
+            drop_in_text += f'[{section}]\n'
+            for setting, values in settings.items():
+                drop_in_text += '\n'.join(
+                    [f'{setting}={value}' for value in values]
+                )
+            drop_in_text += '\n'
+        return drop_in_text
+
+    def write_drop_in_to_filehandler(self, fh: IO) -> None:
+        if not self:
+            # nothing to write out
+            return
+        drop_in_contents = self.to_str()
+        fh.write(drop_in_contents)
+
+    def __bool__(self) -> bool:
+        if not self.contents or not any(
+            self.contents[key] for key in self.contents.keys()
+        ):
+            # no contents
+            return False
+        return True
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, SystemdDropIn):
+            return False
```

Comment:
`return NotImplemented` : https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror & https://docs.python.org/3.6/library/constants.html#NotImplemented